sdk/resourcemanager/Azure.ResourceManager.Core/docs/MigratingFromCurrentToPreview.md

```diff
@@ -0,0 +1,86 @@
+
+You can usually tell by the `id` string itself which type it is. If you're unsure, you can cast it onto a **ResourceIdentifier** and use the `Try` methods to retrieve the values.
+
+### Casting to a specific type
+```csharp
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet";
+// We know the subnet is a resource group level identifier since it has a resource group name in its string
+ResourceGroupResourceIdentifier id = resourceId;
+Console.WriteLine($"Subscription: {id.SubscriptionId}");
+Console.WriteLine($"ResourceGroup: {id.ResourceGroupName}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
+
+### Casting to the base resource identifier
+```csharp
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet";
+// Assume we don't know what type of resource id we have we can cast to the base type
+ResourceIdentifier id = resourceId;
+string property;
+if (id.TryGetSubscriptionId(out property))
+    Console.WriteLine($"Subscription: {property}");
+if (id.TryGetResourceGroupName(out property))
+    Console.WriteLine($"ResourceGroup: {property}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
+
+## [Resource]Operations
+
+This represents a service client that is scoped to a particular resource.
+You can directly execute all operations on that client without needing to pass in scope
+parameters such as subscription id or resource name.
+
+*Old*
+```csharp
+    ComputeManagementClient computeClient = new ComputeManagementClient(subscriptionId, new DefaultAzureCredential());
+    string rgName = "myRgName";
+    string vmName = "myVmName";
+
+    // Each method we call needs to have the scope passed in to know which vm to operate on
+    await computeClient.StartPowerOff(rgName, vmName).WaitForCompletionAsync();
+    await computeClient.StartStart(rgName, vmName).WaitForCompletionAsync();
+```
+
+*New*
+```csharp
+    string rgName = "myRgName";
+    string vmName = "myVmName";
+    Subscription subscription = armClient.DefaultSubscription;
+    ResourceGroup resourceGroup = await subscription.GetResourceGroups().GetAsync(rgName);
```

Comment:
Why does the second example get the vm details before performing any operations?